py-polars/tests/unit/series/test_series.py

```diff
@@ -117,6 +121,24 @@ def test_init_inputs(monkeypatch: Any) -> None:
     # lists
     assert pl.Series("a", [[1, 2], [3, 4]]).dtype == pl.List(pl.Int64)
 
+    # conversion of Date to Datetime
+    s = pl.Series([date(2023, 1, 1), date(2023, 1, 2)], dtype=pl.Datetime)
+    assert s.to_list() == [datetime(2023, 1, 1), datetime(2023, 1, 2)]
+    assert Datetime == s.dtype
+    assert s.dtype.time_unit == "us"  # type: ignore[union-attr]
+    assert s.dtype.time_zone is None  # type: ignore[union-attr]
+
+    # conversion of Date to Datetime with specified timezone and units
+    tu: Final = "ms"
+    tz: Final = "America/Argentina/Rio_Gallegos"
```

Comment:
final looks a bit odd here 😄 I think it's usually used in classes to indicate that things shouldn't be overridden, can we remove it?

Reply:
The `Final` keyword is because the signature for `pl.Date`:

```python
def datetime_(
    ...
    time_unit: TimeUnit = "us",
    ...
) -> Expr:
```
includes the `TimeUnit` type alias which is defined as  `TimeUnit: TypeAlias = Literal["ns", "us", "ms"]`. MyPy doesn't accept strings as literals unless they are labeled as `Final`.

I can remove it from the `tz`, however, or I can put a `# type:ignore` in there if you want to clean up the `Final` stuff.

Reply:
Does it work to annotate it as `TimeUnit` instead?

Reply:
Worked like a charm! Thank you.